File: 4673.cpp
#include <stdio.h>

int cache[10001];

void d(int n){
	int result = n;
	while(n > 0){
		result += n % 10;
		n /= 10;
	}
	if(result > 10000) return;
	cache[result] = 1;
	return d(result);
}

int main(){
	for(int i = 1; i <= 10000; i++) d(i);
	for(int i = 1; i <= 10000; i++){
		if(!cache[i]) printf("%d\n", i);
	}
	return 0;
}

File: 4673.py
cache = [];

def d(n):
	result = n;
	while(n > 0):
		di = int(n % 10)
		result = result + di;
		n = int(n / 10);

	if(result > 10000):
		return
	if not (result in cache):
		cache.append(result);
	return d(result);


for i in range(1, 10000):
	d(i)
for i in range(10000):
	if not i in cache:
		print(i)